Expected errors 2053

ZuleykhaPavlichenkova-TomTom opened the pull request:
This PR adds REGEX error messages to the test cases according to the https://github.com/duckdblabs/duckdb-internal/issues/2053#event-12874954527:

- test/sql/types/decimal/decimal_try_cast.test
- test/sql/types/interval/interval_constants.test
- test/sql/types/interval/interval_try_cast.test
- test/sql/types/unsigned/test_unsigned_arithmetic.test
- test/sql/types/unsigned/test_unsigned_auto_cast.test
- test/sql/catalog/test_if_not_exists.test
- test/sql/types/interval/test_interval.test

This PR also makes `result.py` matching regex logic be the same as in the `result_helper.cpp`.
The change should handle the case when a test provides `<!REGEX>` condition, but a string provided as an error message is a substring of the valid error message, like in this [CI failure](https://github.com/duckdb/duckdb/actions/runs/11171873219/job/31057417988)

ZuleykhaPavlichenkova-TomTom committed:
Add expected error message to test/sql/types/unsigned/test_unsigned_auto_cast.test

ZuleykhaPavlichenkova-TomTom committed:
Add expected regex message to test/sql/types/unsigned/test_unsigned_arithmetic.test

ZuleykhaPavlichenkova-TomTom committed:
Add expected regex message to test/sql/types/interval/interval_constants.test

ZuleykhaPavlichenkova-TomTom committed:
Add expected regex message to test/sql/types/interval/interval_try_cast.test

ZuleykhaPavlichenkova-TomTom committed:
Add expected regex message to test/sql/catalog/test_if_not_exists.test

Tmonster commented on scripts/sqllogictest/result.py:
```diff
@@ -447,10 +447,11 @@ def matches_regex(input: str, actual_str: str) -> bool:
-    if regex_str[-2:] != '.*':
-        regex_str = regex_str + '.*'
+    if should_match:
+        if regex_str[:2] != '.*':
```
I don't think this is the correct way to fix this. This says, "If there should be a match, find the match as a substring or as the whole string".
This can also then mean, if there should not be a match, treat the regex string as is. This could also mean, the regex string must match the actual string completely. 

The regex string should be treated the same regardless of if there should be a match or not. I've tried to find a counter example, but in this exact scenario it's hard

We can talk about this after lunch

hmeriann replied:
Thank you for the detailed expansion!

Tmonster reviewed (commented):
Thanks,  just the one comment

ZuleykhaPavlichenkova-TomTom committed:
Add expected regex message to test/sql/catalog/test_if_not_exists.test

ZuleykhaPavlichenkova-TomTom committed:
make result.py matching regex logic be the same as in the result_helper.cpp

ZuleykhaPavlichenkova-TomTom committed:
Add expected regex message to test_interval.test

Tmonster commented on scripts/sqllogictest/result.py:
```diff
@@ -448,10 +448,10 @@ def matches_regex(input: str, actual_str: str) -> bool:
-        regex_str = regex_str + '.*'
-
+        regex_str = regex_str + '$'
+        
```
Also could you used just `"string"` or just `'string'`? Think it's nicer if it is consistent.

Tmonster reviewed (approved):
Looks good to me except for the code quality

ZuleykhaPavlichenkova-TomTom committed:
Add missing symbols

ZuleykhaPavlichenkova-TomTom committed:
format check

ZuleykhaPavlichenkova-TomTom committed:
Merge branch 'main' into expected-errors-2053

ZuleykhaPavlichenkova-TomTom committed:
Merge remote-tracking branch 'upstream/main' into expected-errors-2053

ZuleykhaPavlichenkova-TomTom committed:
restore main version of the scripts/sqllogictest/result.py

ZuleykhaPavlichenkova-TomTom committed:
Merge branch 'main' into expected-errors-2053

ZuleykhaPavlichenkova-TomTom committed:
add expecter REGEX to
test/common/test_cast_struct.test
test/extension/autoloading_base.test
test/extension/autoloading_current_setting.test
test/extension/autoloading_encodings.test
test/extension/autoloading_load_only.test

hannes commented:
Thanks!
